### ISSUE TITLE
Trying to repack 'jslint' doesn't work

### ISSUE THREAD

**Issue by illuusio** [marguerite/nodejs-packaging#3]
Trying to pack 'jslint' with new Nova tool throws error:
<code>./nova jslint</code>

<pre>
["read-pkg", "load-json-file", "pinkie-promise"]
1:jslint
2:exit
3:glob
4:inflight
5:once
6:wrappy
7:wrappy
8:inherits
9:minimatch
10:brace-expansion
11:balanced-match
12:concat-map
13:once
14:wrappy
15:nopt
16:abbrev
/something/tool/dependencies.rb:77:in `list': undefined method `[]' for nil:NilClass (NoMethodError)
    from /something/nodejs-packaging/tool/dependencies.rb:79:in `block in list'
    from /something/nodejs-packaging/tool/dependencies.rb:78:in `each'
    from /something/nodejs-packaging/tool/dependencies.rb:78:in `list'
    from/something/nodejs-packaging/tool/dependencies.rb:79:in `block in list'
    from /something/nodejs-packaging/tool/dependencies.rb:78:in `each'
    from/something/nodejs-packaging/tool/dependencies.rb:78:in `list'
    from /something/nodejs-packaging/tool/dependencies.rb:110:in `write'
    from ./nova:12:in `<main>'
</pre>

**Comment by marguerite** [comment]
Thanks! gulp is facing the same problem. That both 'load-json-file' and 'path-type' contain 'pinkie-promise'. 

My parent.rb didn't consider that situation in advance. I need to reimplement the parent finding logic.

And please don't leave me alone on this toolkit. I just caught a cold yesterday. So sorry for the bad manners.

**Comment by illuusio** [comment]
I think I just get over reacting with this but I change also my packing system to be more agile.. old one was pain in the ar*se anyways. Keep on working with this and it will be good.
What should nova do after it have downloaded packages because with me it just hangs..

**Comment by marguerite** [comment]
@illuusio just hit enter :-) it does nothing after downloading the packages.

**Comment by marguerite** [comment]
@illuusio 

Now `nova` works for `jslint`! (I just tested it!)

Sorry the reimplementation of json parent finding took time that is more than expected. but well I'd say the result is also more than expected!. I have tested the new logic on `gulp`, `read-pkg-up`, `phantomjs-prebuilt`. the former two contains multiple `pinkie-promise` or `normalize-package-data` on different tree branches. the later is for the comparison to see if it still works for the simple case.

Issue closed and thanks again. I'm glad I can have such a valuable tester who can maintain 800+ packages from the start :-)
